mailpoet/tests/DataGenerator/Generators/WooCommercePastRevenues.php

```diff
@@ -343,55 +367,57 @@ private function createProduct($name, $price, $categoryIds = [], $discount = 0)
   /**
    * @return array
    */
-  private function createSentEmailData(NewsletterEntity $newsletter, $sentAt, $subscribersIds, $segmentId) {
+  private function createSentEmailData(NewsletterEntity $newsletter, string $sentAt, array $subscribersIds, $segmentId) {
+    $connection = $this->entityManager->getConnection();
+
     // Sending task
-    $task = ScheduledTask::createOrUpdate([
-      'type' => Sending::TASK_TYPE,
-      'status' => ScheduledTask::STATUS_COMPLETED,
-      'created_at' => $sentAt,
-      'processed_at' => $sentAt,
-    ]);
-    $task->save();
+    $task = new ScheduledTaskEntity();
```

Comment:
`MailPoet\Test\DataFactories\ScheduledTask` can facilitate creation

Reply:
Please correct me if I'm wrong but I don't think there is a way to set `createdAt` and `processed` using the ScheduledTask data factory. We could add it, but I would suggest we leave this for another PR as the current approach should work fine.

Reply:
I'm fine with leaving it to another PR, I only suggested doing it here since thought adding the missing part to `MailPoet\Test\DataFactories\ScheduledTask` are not very time intensive, and I leave this decision to you.